server/modules/protocol/MySQL/mariadbbackend/mysql_backend.c

```diff
@@ -977,6 +977,8 @@ gw_read_and_write(DCB *dcb)
                 gwbuf_set_type(stmt, GWBUF_TYPE_RESULT);
             }
 
+            mxs_mysql_get_session_track_info(stmt, proto->server_capabilities);
```

Comment:
This call should be done on line 762 inside `modutil_get_complete_packets` as the first packet is not guaranteed to be a OK packet. For example, a stored procedure call can return multiple result sets followed by a single OK packet. Each OK packet needs to be inspected to know the correct transaction state. Doing the processing in `modutil_get_complete_packets` would allow the packets to be detected correctly.

Reply:
is ok to call mxs_mysql_get_session_track_info after modutil_get_complete_packets, because server_capabilitiy is need to parse ok packet;